app/components/pool/pool-endpoint/inbound-nat-pool-picker.component.ts

```diff
@@ -0,0 +1,122 @@
+import { Component, Input, OnDestroy, forwardRef } from "@angular/core";
+import {
+    ControlValueAccessor, FormBuilder, FormControl, FormGroup, NG_VALIDATORS, NG_VALUE_ACCESSOR, Validator, Validators,
+} from "@angular/forms";
+import { Subscription } from "rxjs";
+
+import { InboundEndpointProtocol, InboundNATPool } from "app/models";
+import "./inbound-nat-pool-picker.scss";
+import * as EndpointHelper from "./pool-endpoint-helper";
+
+@Component({
+    selector: "bl-inbound-nat-pool-picker",
+    templateUrl: "inbound-nat-pool-picker.html",
+    providers: [
+        // tslint:disable:no-forward-ref
+        { provide: NG_VALUE_ACCESSOR, useExisting: forwardRef(() => InboundNATPoolPickerComponent), multi: true },
+        { provide: NG_VALIDATORS, useExisting: forwardRef(() => InboundNATPoolPickerComponent), multi: true },
+    ],
+})
+export class InboundNATPoolPickerComponent implements ControlValueAccessor, Validator, OnDestroy {
+    @Input() public inboundNATPools: InboundNATPool[];
+
+    public InboundEndpointProtocol = InboundEndpointProtocol;
+    public minPort = EndpointHelper.MININUM_PORT;
```

Comment:
can't you just pass the EndpointHelper here instead of all those values, also I feel like we don't really name our constant ALL_CAPS so not sure if we should keep it like that